### PR TITLE
Modular components

### PR DESCRIPTION
**Pull request by Tithen-Firion**
### Warframe Hub Pull Request
---
---

**Summary (short):** rewrite components code to modular form

---
**Description:** better readability, easier to extend, will allow us to shorten some other code

---
**Fixes issue (include link):**

---
**Mockups, screenshots, evidence:**

---

(Check one)
- [ ] Issue fix
- [x] Suggestion fulfillment

### CODE REVIEW

**Comment by TobiTenno** [comment]
Seems ok at first ui usage, i'll do a code dive later

**Comment by Tithen-Firion** [comment]
First things I saw now:
- there's no space between archwing and nightmare alert icons but that can be fixed easily
- their order is not the same as in current version - want me to swap it?

Alert rewards are formatted the same way as invasion ones. So `1 Nitain Extract` becomes just `Nitain Extract`. Don't know if you want to keep it that way or not.

**Comment by TobiTenno** [comment]
yeah, i'm fine with the removal of the `1 ` at the start of strings.
i'd definitely fix the space. 
which are misplaced?

**Comment by Tithen-Firion** [comment]
There was an alert that was both archwing and nightmare. Expired already. Currently those alerts have archwing icon and then nightmare. In this PR they are reversed.

**Comment by TobiTenno** [comment]
oh, the icons. that's fine, i don't mind either way. the order was just cause i threw it someplace, and wasn't in a particular order

**Comment by EricSihaoLin** [comment]
Is the plan to completely migrate update functions to the new component layout?

File: assets/js/main.js [hunk 1]
```diff
@@ -845,8 +866,74 @@ function removeTimeBadgeColor(element) {
   element.removeClass('label-info');
 }
 
+// helper function, transforms miliseconds diff into one of:
+// Dd HHh MMm SSs
+// Hh MMm SSs
+// Mm SSs
+// Ss
```
Comment (TobiTenno) [inline assets/js/main.js]: why are there like 5 lines comments for this? do it in a block.

File: views/partials/templates.hbs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<div id="templates" class="hide">
+	<li id="alerts-template" class="list-group-item list-group-item-borderless">
```
Comment (TobiTenno) [inline views/partials/templates.hbs]: if you're going for full modular, why not make each a partial?

**Comment by TobiTenno** [comment]
@EricSihaoLin that's my impression

File: views/partials/templates.hbs (hunk 2 above)
Comment (Tithen-Firion) [reply views/partials/templates.hbs]: Actually I wanted to go even further and pack those component modules like this:
```
├───assets
│  ...
├───component_modules
│   ├───alerts
│   │   ├───component.js
│   │   └───template.hbs
│  ...
│   └───invasions
│       ├───component.js
│       └───template.hbs
...
```
That would require `gulp-concat` for sure.

Comment (TobiTenno) [reply views/partials/templates.hbs]: i mean, i think it'd make it easier in the long run for us, go for it

**Comment by Tithen-Firion** [comment]
I'm using them in other files. I moved invasion-specific functions to invasion component but those are shared between components and used by components only.

File: assets/js/main.js [hunk 3]
```diff
@@ -660,139 +588,41 @@ function updateNews() {
...
-  case 'Infested':
-    return 'infested-invasion';
-  case 'Corrupted':
-    return 'corrupted-invasion';
-  default:
-    return 'default-invasion';
+  for (const countedItem of rewardsData.countedItems) {
+    const count = countedItem.count > 1 ? `${countedItem.count} ` : '';
+    rewards.push(count + countedItem.type);
+    // rewardsSet.add(countedItem.type);
```
Comment (TobiTenno) [inline assets/js/main.js:557]: could we get rid of the random commented code?

**Comment by TobiTenno** [comment]
looks good, lets continue this way. once the templates are all made, i'd like to see that in a loop instead of a bunch of template calls. 

might want to pull those as a list of directory names if we wanna keep it really quick to add new ones, but i'm fine with a static array if that works best

File: assets/js/main.js (hunk 3 above)
Comment (Tithen-Firion) [reply assets/js/main.js:557]: Sure we can, just wanted to leave it for when we decide to implement reward filtering.

**Comment by Tithen-Firion** [comment]
I don't quite get what static array you're referring to.

**Comment by TobiTenno** [comment]
It doesn't exist yet... I'm referring to once everything is templated.

**Comment by TobiTenno** [comment]
closing in favor of the more over-arching #202 project

**Pull request closed without merging**